src/hex/shape_constructors.rs

```diff
@@ -0,0 +1,202 @@
+use std::error::Error;
```

Comment:
I'm thinking that given 'constructor' has a very specific meaning in OO programming, does 'builder' make more sense as a term that describes these clearly (as in the builder pattern, ofc) without that meaning.